### DISCUSSION TITLE
Automatic language detection

### DISCUSSION THREAD

**Discussion by sangyuxiaowu** [massCodeIO/massCode#158] [Ideas]
One thing that feels troublesome to use is that the language needs to be manually selected and changed, and then the drop-down box is too long, and there is no search, it takes a long time to search, and it is dazzling.

Vscode provides a good [program reference](https://code.visualstudio.com/updates/v1_60#_automatic-language-detection) . 

The automatic language detection feature uses machine learning to guess the language and the machine learning model runs entirely local on your machine. The model is powered by the open-source ML library, [Tensorflow.js](https://www.tensorflow.org/js/), and the ML model from [Guesslang](https://github.com/yoeo/guesslang).

**Comment by antonreshetov** [comment]
I was already thinking about it, did some research and came across https://github.com/microsoft/vscode-languagedetection. This tool uses SnippetsLab. 

I have used this feature in SnippetsLab a little bit, not that I like it, I mean the idea in general. There are wrong identifications of the language. 

I will keep thinking in that direction. 

P.S. Search is available natively in OS just start typing the desired language when the selector is open.
P.S.S. You can also set the default language to the folder.

**Reply by sangyuxiaowu** [reply]
Thanks for reminding. I just found that I can really search the select component.
